### PR TITLE
Feature: request parser

### PR DESCRIPTION
**Pull request by hosseinsokhandan**
This PR introduces a request parser for HTTP 1.0 and HTTP 1.1:
- headparser - header parser
- reqtok - request parser for `verb`, `path`, and `version`.
- reqpaser - parse request

### CODE REVIEW

**Comment by hosseinsokhandan** [comment]
This PR is marked as Draft as its failure tests are not implemented yet.
I would appreciate it if you could take a look at it.

CC: @pylover

**Review by pylover — APPROVED** [review]
:100:

File: tests/test_request_parse.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+#include <cutest.h>
+#include <clog.h>
+
+
+struct chttpd_request {
+    char *verb;
+    char *path;
+    char *version;
+    char *content_length;
```
Comment (pylover) [inline tests/test_request_parse.c:29]: As mentioned before, content length should be integer.

File: tests/test_request_parse.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    char *content_type;
+    char *connection;
+    char *accept;
+    char *httpheader;
+};
+
+
+void
+trim(char **str) {
```
Comment (pylover) [inline tests/test_request_parse.c:38]: Use signle `*str` instead of `**str`.

File: tests/test_request_parse.c [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    struct chttpd_request conn;
+
+    eqint(0, request_parse(request, &conn));
+    eqstr("GET", conn.verb);
+    eqstr("/foo/bar", conn.path);
+    eqstr("1.1", conn.version);
+    eqstr(conn.accept, "baz");
+    eqstr(conn.connection, "corge");
+    eqstr(conn.content_type, "qux/quux");
+    isnull(conn.content_length);
```
Comment (pylover) [inline tests/test_request_parse.c:116]: Write more tests for content length.

**Review by pylover — CHANGES_REQUESTED** [review]

File: tests/test_request_parse.c (hunk 1 above)
Comment (hosseinsokhandan) [reply tests/test_request_parse.c]: Resolved.

File: tests/test_request_parse.c (hunk 2 above)
Comment (hosseinsokhandan) [reply tests/test_request_parse.c]: Done.

File: tests/test_request_parse.c (hunk 3 above)
Comment (hosseinsokhandan) [reply tests/test_request_parse.c]: Resolved.

**Review by pylover — APPROVED** [review]

**Pull request merged**